### PR TITLE
update deps, fix role model

### PR DESCRIPTION
**Pull request by delebash**
update npm and composer deps, fixed pagination import for updated dep, added spatie role since role model was removed.

### CODE REVIEW

File: database/seeders/CreateAdminUserSeeder.php [hunk 1]
```diff
@@ -32,5 +32,19 @@ public function run()
         $role->syncPermissions($permissions);
 
         $user->assignRole([$role->id]);
+
+        $role2 = Role::create(['name' => 'super-admin']);
+        // gets all permissions via Gate::before rule; see AuthServiceProvider
+        // create demo users
+        $user = User::create([
```
Comment (irabbi360) [inline database/seeders/CreateAdminUserSeeder.php:39]: Why need to super admin gate user role? Already have an admin role. So no need to add a super admin role from the seed. If you need it you can create it from the admin panel & assign it to a user.

File: package-lock.json
Comment (irabbi360) [inline package-lock.json:1]: don't push package-lock file. add in on gitignore.

File: database/seeders/CreateAdminUserSeeder.php (hunk 1 above)
Comment (delebash) [reply database/seeders/CreateAdminUserSeeder.php:39]: ok, that make sense.

File: package-lock.json
Comment (delebash) [reply package-lock.json:1]: done

Comment (nabin009) [reply package-lock.json:1]: permission model is missing and authserviceprovider check the permission model

**Comment by nabin009** [comment]
permission model is missing and authserviceprovider check the permission model

**Comment by delebash** [comment]
I thought since we are using spatie that we just use the spatie permission and role models without having to create one?  Also, I couldn't get the role-permssion I added to work with composable so maybe you can take a look at it.  In the admin/roles/edit you can see the code I added in the setup just calling the axios directly and binding it to the listbox control.

**Comment by irabbi360** [comment]
You updated this 'casl/ability' plugin. So also you need to update the ability service. In the new version, a few things are deprecated.
image

Please also update the readme doc for how to visit the request-docs URL. Also if possible please update the URL '/api-docs' instead of '/request-docs'.

**Comment by delebash** [comment]
update ability changed api docs url

**Comment by delebash** [comment]
Also, I added a restriction where a user can only edit their own posts unless the role is admin.  Maybe it would be better to add a new permission for edit all posts and edit own posts instead of restricting by role.

**Comment by irabbi360** [comment]
In the posts, you added multiple categories of support. So, now you should also need to update the admin posts list page for the multiple category-wise filters & display the category name.

**Comment by delebash** [comment]
I am having trouble with the category search.  I think my relations are setup correctly but when I try to search by category the sql statement is wrong. Hardcoded as 1 for testing

PostController

 $posts = Post::whereHas('categories')
            ->when(request('search_category'), function ($query) {
                $query->where('categories.id',1);
            })

sql error:

Column not found: 1054 Unknown column 'categories.id' in 'where clause' (Connection: mysql, SQL: select count(*) as aggregate from `posts` where exists (select * from `categories` inner join `category_post` on `categories`.`id` = `category_post`.`category_id` where `posts`.`id` = `category_post`.`post_id`) and `categories`.`id` = 1)"

**Comment by delebash** [comment]
So this works but I don't really know why or how to get it in the query syntax

  $posts = Post::with('categories')
            ->whereHas('categories', function ($query) {
                $query->where('categories.id', 'like', '%3%');
            })
//            ->when(request('search_category'), function ($query) {
//                $query->where('categories.id',1);
//            })

**Comment by delebash** [comment]
This seems to work, let me know if it is ok

$posts = Post::whereHas('categories')
            ->whereHas('categories', function ($query) {
                if(request('search_category')){
                    $query->where('categories.id',3);
                }
            })

**Comment by delebash** [comment]
ok I think I have everything working and updated.  Let me know.

**Comment by delebash** [comment]
I added spatie media for image upload, that works fine.  However, I can't retrieve the uploaded image related to the post model.  The file is saved on disk and the correct entry is added to the media table. But thumbnail just returns an empty array. Can you help?

PostController 

 $posts = Post::with('media')

PostResoure
'thumbnail' => $this->media

postman query result

```
 "data": [
        {
            "id": 37,
            "title": "dfgdfgsdfgdsgsdfgsdfg",
            "categories": [
                {
                    "id": 1,
                    "name": "Vue.js",
                    "created_at": "2023-10-02T17:44:52.000000Z",
                    "updated_at": "2023-10-02T17:44:52.000000Z",
                    "pivot": {
                        "post_id": 37,
                        "category_id": 1
                    }
                },
                {
                    "id": 2,
                    "name": "Cat 2",
                    "created_at": "2023-10-02T17:44:52.000000Z",
                    "updated_at": "2023-10-02T17:44:52.000000Z",
                    "pivot": {
                        "post_id": 37,
                        "category_id": 2
                    }
                }
            ],
            "content": "sdfsdfgsdfgdsfgsdfgsdfgsdfgsdfgsdfgsdfgsdfgsdfsdfg...",
            "thumbnail": [],
            "created_at": "2023-10-02"
        }
```

**Comment by irabbi360** [comment]
read this doc
https://spatie.be/docs/laravel-medialibrary/v10/basic-usage/retrieving-media

image

image

**Review by irabbi360 — APPROVED** [review]

**Pull request merged**